Osgis.bundleFinder: support version range

aledsage opened the pull request:
Useful for `brooklyn.libraries` when name:version of pre-installed bundle is specified.

@m4rkmckenna as discussed - can you take a look please?

m4rkmckenna commented on core/src/test/java/org/apache/brooklyn/util/core/osgi/OsgisTest.java:
```diff
@@ -0,0 +1,117 @@
+                .symbolicName("mybundle")
+                .version("[1.0.0, 2.0.0)")
+                .findAll();
+        assertEquals(result2, ImmutableList.of(myBundle1_0_0, myBundle1_1_0));
```
I think its worth adding the versions `2.0.0.SNAPSHOT` & `2.0.0` and making assertions around this boundary

Alos add checks for the range `[1.0.0, 2.0.0]`

aledsage commented:
Thanks @m4rkmckenna - I've added a couple more tests, squashed it into a single commit, and rebased against master. Once jenkins confirms this builds, I'll merge it.
